src/async_fuse/memfs/dist/lock_manager.rs

```diff
@@ -0,0 +1,384 @@
+        tokio::spawn(async move {
+            let mut futures = FuturesUnordered::new();
+
+            loop {
+                select! {
+                    // 1. new lock renew job
+                    new_lock= new_lock_rx.recv() => {
+                        if let Some(new_lock)=new_lock{
+                            futures.push(Self::renew_lock_future(new_lock, &fs_async_sender));
+                        }
+                    },
+                    // 2. poll one renew job
+                    continue_renew_lock= futures.next() => {
+                        if let Some(Some(continue_renew_lock))=continue_renew_lock{
+                            futures.push(Self::renew_lock_future(continue_renew_lock, &fs_async_sender));
+                        }
+                    },
+                    // 3. system end
+                    _ = &mut system_end_rx => {
+                        break;
+                    }
+                }
+            }
+        })
+    }
+
+    /// new `DistRwLockManager`
+    pub fn new(
+        kv_engine: Arc<KVEngineType>,
+        fs_async_sender: FsAsyncResultSender,
+    ) -> (Arc<DistLockManager>, JoinHandle<()>) {
+        //init a main loop to renew the lock lease
+        let (register_renew_tx, register_renew_rx) =
+            tokio::sync::mpsc::channel::<LockDescription>(10);
+        let (system_end_tx, system_end_rx) = tokio::sync::oneshot::channel::<()>();
+        let renew_task =
+            Self::spawn_renew_lock_task(register_renew_rx, system_end_rx, fs_async_sender);
+        let man: Arc<DistLockManager> = Arc::new(DistLockManager {
+            locked: RwLock::new(DistLockManagerLocked {
+                timeout_renews: HashMap::new(),
+                system_end_tx: Some(system_end_tx),
+            }),
+            kv_engine,
+            register_renew_tx,
+        });
+
+        (man, renew_task)
+    }
+
+    /// try lock dist lock
+    /// - return true if lock success
+    #[inline]
+    pub async fn try_lock(&self, key: &LockKeyType) -> DatenLordResult<bool> {
+        self.inner_lock(key, true).await
+    }
+
+    #[inline]
+    /// lock dist lock
+    pub async fn lock(&self, key: &LockKeyType) -> DatenLordResult<()> {
+        self.inner_lock(key, false).await.map(|_| ())
+    }
+    /// jnner lock function for lock and try lock
+    /// todo: fix bug, when wating for the lock, lease should be updated.
+    async fn inner_lock(&self, key: &LockKeyType, try_lock: bool) -> DatenLordResult<bool> {
+        log::debug!("`LockManager::lock`, key: {key:?}");
+        let key_vec: Vec<u8> = key.get_key();
+
+        // already locked, only need the system to renew the lease
+        if self.locked.read().timeout_renews.contains_key(&key_vec) {
+            return Ok(true);
+        }
+        let lease_info = self.kv_engine.lease_grant(DEFAULT_TIMEOUT).await?;
+        // lock
+        let lockinfo = if try_lock {
+            match tokio::time::timeout(
```

Comment:
When locking a key in etcd, a k/v pair is generated which can be retrieved using 'get'. If the key already exists, 'try_lock' will return immediately. This can reduce the delay caused by 'try_lock'.

Reply:
<img width="620" alt="image" src="https://github.com/datenlord/datenlord/assets/8100696/07a9def8-52db-488f-aced-0bc84ab4522a">

Later we can test the self implemented rwlock's performance, if it's better we can replace the native lock with it.